dpnp/backend/kernels/dpnp_krnl_arraycreation.cpp

```diff
@@ -128,20 +128,45 @@ void dpnp_ones_like_c(void* result, size_t size)
 }
 
 template <typename _DataType>
-void dpnp_zeros_c(void* result, size_t size)
+class dpnp_tri_c_kernel;
+
+template <typename _DataType>
+void dpnp_tri_c(void* result1, const size_t N, const size_t M, const int k)
 {
-    _DataType* fill_value = reinterpret_cast<_DataType*>(dpnp_memory_alloc_c(sizeof(_DataType)));
-    fill_value[0] = 0;
+    cl::sycl::event event;
 
-    dpnp_initval_c<_DataType>(result, fill_value, size);
+    _DataType* result = reinterpret_cast<_DataType*>(result1);
 
-    dpnp_memory_free_c(fill_value);
-}
+    size_t idx = N* M;
```

Comment:
Will your algo work correct if N and M multiplication is zero?
Do you have any tests for this case?